datastream/client.go

```diff
@@ -411,7 +411,7 @@ func (c *DataStreamClient) handleErrorMessage(_ context.Context, resp *DataStrea
 	return errors.New(fmt.Sprintf("%s", respError.Error))
 }
 
-func (c *DataStreamClient) CheckFlag(ctx context.Context, evalCtx *schematicgo.CheckFlagRequestBody, flagKey string) bool {
+func (c *DataStreamClient) CheckFlag(ctx context.Context, evalCtx *schematicgo.CheckFlagRequestBody, flagKey string) *rulesengine.CheckFlagResult {
```

Comment:
We don't expect any end users to be calling this function directly right? So the return type change here shouldn't matter?